onnxruntime/python/onnxruntime_pybind_mlvalue.cc

```diff
@@ -266,7 +277,7 @@ std::unique_ptr<Tensor> CreateTensor(const AllocatorPtr& alloc, const std::strin
   TensorShape shape = GetArrayShape(darray);
   auto element_type = NumpyToOnnxRuntimeTensorType(npy_type);
   if (npy_type != NPY_UNICODE && npy_type != NPY_STRING &&
-      npy_type != NPY_VOID && npy_type != NPY_OBJECT) {
+      npy_type != NPY_VOID && npy_type != NPY_OBJECT && use_numpy_data_memory) {
```

Comment:
Can you use IsNumericNumpyType?

Reply:
Ideally, yes. This check here (without the flag I added) basically forms the core check in `IsNumericNumpyType`. I sill have one pending comment there - to look into the check you suggested.

Reply:
Made this check IsNumericNumpyType(npy_type) && use_numpy_data_memory